internal/resolution/datasource/npm_registry_test.go

```diff
@@ -0,0 +1,141 @@
+package datasource_test
+
+import (
+	"context"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+	"slices"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/google/osv-scanner/internal/resolution/datasource"
+	"github.com/google/osv-scanner/internal/testutility"
+)
+
+type fakeNpmRegistry struct {
```

Comment:
This is practically the same as the mock http server in the maven tests, and it's probably going to be used in other tests.
I'm going to add a more general mock http server to testutility in another PR

Reply:
that would be nice! I definitely implement this again for Maven parent support....